{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "1ea71dc2-b259-4526-aa6c-ca43d424c6e8",
   "metadata": {},
   "source": [
    "# ICMR Antimicrobial Resistance Project\n",
    "\n",
    "Date : 08-02-2025"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "acdd6131-a8eb-4083-bbb0-f4819e77f2d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# importing necessry libraries\n",
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "import seaborn as sns\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "32a9b63b-827a-4dbf-8510-791407ca5d94",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>suv1</th>\n",
       "      <th>suv2</th>\n",
       "      <th>suv3</th>\n",
       "      <th>suv4</th>\n",
       "      <th>suv5</th>\n",
       "      <th>suv6</th>\n",
       "      <th>suv7</th>\n",
       "      <th>suv8</th>\n",
       "      <th>suv9</th>\n",
       "      <th>suv10</th>\n",
       "      <th>...</th>\n",
       "      <th>hv5</th>\n",
       "      <th>hv6</th>\n",
       "      <th>hv7</th>\n",
       "      <th>hv8</th>\n",
       "      <th>hv9</th>\n",
       "      <th>hv10</th>\n",
       "      <th>sv1</th>\n",
       "      <th>sv2</th>\n",
       "      <th>sv3</th>\n",
       "      <th>sv4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>327514</td>\n",
       "      <td>3616396</td>\n",
       "      <td>312357</td>\n",
       "      <td>309247</td>\n",
       "      <td>309252</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "      <td>1.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>39</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>URINARY TRACT INFECTION WITH RENAL CALCULI</td>\n",
       "      <td>,CHRONIC KINDEY DISEASE</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>2021-01-01</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>327514</td>\n",
       "      <td>3616397</td>\n",
       "      <td>312357</td>\n",
       "      <td>309247</td>\n",
       "      <td>309252</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "      <td>1.0</td>\n",
       "      <td>95.0</td>\n",
       "      <td>39</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>URINARY TRACT INFECTION WITH RENAL CALCULI</td>\n",
       "      <td>,CHRONIC KINDEY DISEASE</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>2021-01-01</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>327514</td>\n",
       "      <td>2730264</td>\n",
       "      <td>312357</td>\n",
       "      <td>309247</td>\n",
       "      <td>309252</td>\n",
       "      <td>31.0</td>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>URINARY TRACT INFECTION WITH RENAL CALCULI</td>\n",
       "      <td>,CHRONIC KINDEY DISEASE</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>2021-01-01</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>327514</td>\n",
       "      <td>2730267</td>\n",
       "      <td>312357</td>\n",
       "      <td>309247</td>\n",
       "      <td>309252</td>\n",
       "      <td>81.0</td>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>URINARY TRACT INFECTION WITH RENAL CALCULI</td>\n",
       "      <td>,CHRONIC KINDEY DISEASE</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>2021-01-01</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>327514</td>\n",
       "      <td>2730263</td>\n",
       "      <td>312357</td>\n",
       "      <td>309247</td>\n",
       "      <td>309252</td>\n",
       "      <td>22.0</td>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>URINARY TRACT INFECTION WITH RENAL CALCULI</td>\n",
       "      <td>,CHRONIC KINDEY DISEASE</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>2021-01-01</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     suv1     suv2    suv3    suv4    suv5  suv6  suv7  suv8  suv9  suv10  \\\n",
       "0  327514  3616396  312357  309247  309252   NaN    15   1.0  94.0     39   \n",
       "1  327514  3616397  312357  309247  309252   NaN    15   1.0  95.0     39   \n",
       "2  327514  2730264  312357  309247  309252  31.0    15   NaN   NaN     39   \n",
       "3  327514  2730267  312357  309247  309252  81.0    15   NaN   NaN     39   \n",
       "4  327514  2730263  312357  309247  309252  22.0    15   NaN   NaN     39   \n",
       "\n",
       "   ...  hv5                                         hv6  \\\n",
       "0  ...    3  URINARY TRACT INFECTION WITH RENAL CALCULI   \n",
       "1  ...    3  URINARY TRACT INFECTION WITH RENAL CALCULI   \n",
       "2  ...    3  URINARY TRACT INFECTION WITH RENAL CALCULI   \n",
       "3  ...    3  URINARY TRACT INFECTION WITH RENAL CALCULI   \n",
       "4  ...    3  URINARY TRACT INFECTION WITH RENAL CALCULI   \n",
       "\n",
       "                       hv7  hv8  hv9  hv10        sv1  sv2  sv3  sv4  \n",
       "0  ,CHRONIC KINDEY DISEASE    0        NaN 2021-01-01   15    0       \n",
       "1  ,CHRONIC KINDEY DISEASE    0        NaN 2021-01-01   15    0       \n",
       "2  ,CHRONIC KINDEY DISEASE    0        NaN 2021-01-01   15    0       \n",
       "3  ,CHRONIC KINDEY DISEASE    0        NaN 2021-01-01   15    0       \n",
       "4  ,CHRONIC KINDEY DISEASE    0        NaN 2021-01-01   15    0       \n",
       "\n",
       "[5 rows x 34 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load dataset\n",
    "df = pd.read_stata(\"D:\\\\DS & DA Projects\\\\ICMR Antimicrobial Resistance\\\\1730707893_amr_data.dta\",convert_categoricals=False)\n",
    "\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6bdd84a8-d5b2-41ec-80d3-d461cbb94114",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5185, 34)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# total rows and columns\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e2ec217e-4105-4000-8e61-5b54beaa43dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>suv1</th>\n",
       "      <th>suv2</th>\n",
       "      <th>suv3</th>\n",
       "      <th>suv4</th>\n",
       "      <th>suv5</th>\n",
       "      <th>suv6</th>\n",
       "      <th>suv7</th>\n",
       "      <th>suv8</th>\n",
       "      <th>suv9</th>\n",
       "      <th>suv10</th>\n",
       "      <th>...</th>\n",
       "      <th>hv1</th>\n",
       "      <th>hv2</th>\n",
       "      <th>hv3</th>\n",
       "      <th>hv4</th>\n",
       "      <th>hv5</th>\n",
       "      <th>hv8</th>\n",
       "      <th>hv10</th>\n",
       "      <th>sv1</th>\n",
       "      <th>sv2</th>\n",
       "      <th>sv3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>5185.000000</td>\n",
       "      <td>5.185000e+03</td>\n",
       "      <td>5185.000000</td>\n",
       "      <td>5185.000000</td>\n",
       "      <td>5.185000e+03</td>\n",
       "      <td>4817.000000</td>\n",
       "      <td>5185.000000</td>\n",
       "      <td>368.000000</td>\n",
       "      <td>368.00000</td>\n",
       "      <td>5185.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>5185.000000</td>\n",
       "      <td>5185</td>\n",
       "      <td>5185.000000</td>\n",
       "      <td>5185.000000</td>\n",
       "      <td>5185.000000</td>\n",
       "      <td>5185.000000</td>\n",
       "      <td>240.000000</td>\n",
       "      <td>5185</td>\n",
       "      <td>5185.000000</td>\n",
       "      <td>5185.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>419914.704532</td>\n",
       "      <td>3.618683e+06</td>\n",
       "      <td>397516.808293</td>\n",
       "      <td>387645.111668</td>\n",
       "      <td>3.012852e+07</td>\n",
       "      <td>30.460868</td>\n",
       "      <td>6.231823</td>\n",
       "      <td>1.426630</td>\n",
       "      <td>50.92663</td>\n",
       "      <td>16.007715</td>\n",
       "      <td>...</td>\n",
       "      <td>8.344841</td>\n",
       "      <td>2022-01-26 12:35:41.253616384</td>\n",
       "      <td>53.827965</td>\n",
       "      <td>2.379942</td>\n",
       "      <td>1.903182</td>\n",
       "      <td>0.205786</td>\n",
       "      <td>1.966667</td>\n",
       "      <td>2022-01-31 20:22:15.853423360</td>\n",
       "      <td>26.572420</td>\n",
       "      <td>0.213308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>327514.000000</td>\n",
       "      <td>2.730262e+06</td>\n",
       "      <td>312357.000000</td>\n",
       "      <td>309247.000000</td>\n",
       "      <td>3.092520e+05</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>2021-01-01 00:00:00</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2021-01-01 00:00:00</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>422638.000000</td>\n",
       "      <td>3.612821e+06</td>\n",
       "      <td>400203.000000</td>\n",
       "      <td>390045.000000</td>\n",
       "      <td>3.148632e+07</td>\n",
       "      <td>11.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>2022-01-18 00:00:00</td>\n",
       "      <td>13.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>2022-01-31 00:00:00</td>\n",
       "      <td>12.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>422784.000000</td>\n",
       "      <td>3.614218e+06</td>\n",
       "      <td>400336.000000</td>\n",
       "      <td>390178.000000</td>\n",
       "      <td>3.148646e+07</td>\n",
       "      <td>20.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>45.00000</td>\n",
       "      <td>12.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>2022-02-12 00:00:00</td>\n",
       "      <td>62.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>2022-02-17 00:00:00</td>\n",
       "      <td>15.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>422941.000000</td>\n",
       "      <td>3.615520e+06</td>\n",
       "      <td>400461.000000</td>\n",
       "      <td>390269.000000</td>\n",
       "      <td>3.148655e+07</td>\n",
       "      <td>45.000000</td>\n",
       "      <td>13.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>94.00000</td>\n",
       "      <td>29.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>14.000000</td>\n",
       "      <td>2022-02-21 00:00:00</td>\n",
       "      <td>72.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>2022-02-23 00:00:00</td>\n",
       "      <td>23.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>423107.000000</td>\n",
       "      <td>4.855579e+06</td>\n",
       "      <td>400602.000000</td>\n",
       "      <td>390395.000000</td>\n",
       "      <td>3.148667e+07</td>\n",
       "      <td>124.000000</td>\n",
       "      <td>33.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>113.00000</td>\n",
       "      <td>114.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>21.000000</td>\n",
       "      <td>2022-02-28 00:00:00</td>\n",
       "      <td>161.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>2022-03-01 00:00:00</td>\n",
       "      <td>83.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>14376.897168</td>\n",
       "      <td>1.677638e+05</td>\n",
       "      <td>13543.280018</td>\n",
       "      <td>12168.351234</td>\n",
       "      <td>6.361543e+06</td>\n",
       "      <td>27.639562</td>\n",
       "      <td>6.889888</td>\n",
       "      <td>0.495261</td>\n",
       "      <td>39.38848</td>\n",
       "      <td>16.938089</td>\n",
       "      <td>...</td>\n",
       "      <td>6.118074</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.748223</td>\n",
       "      <td>0.734079</td>\n",
       "      <td>0.833264</td>\n",
       "      <td>0.404314</td>\n",
       "      <td>0.179881</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.466112</td>\n",
       "      <td>0.409683</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                suv1          suv2           suv3           suv4  \\\n",
       "count    5185.000000  5.185000e+03    5185.000000    5185.000000   \n",
       "mean   419914.704532  3.618683e+06  397516.808293  387645.111668   \n",
       "min    327514.000000  2.730262e+06  312357.000000  309247.000000   \n",
       "25%    422638.000000  3.612821e+06  400203.000000  390045.000000   \n",
       "50%    422784.000000  3.614218e+06  400336.000000  390178.000000   \n",
       "75%    422941.000000  3.615520e+06  400461.000000  390269.000000   \n",
       "max    423107.000000  4.855579e+06  400602.000000  390395.000000   \n",
       "std     14376.897168  1.677638e+05   13543.280018   12168.351234   \n",
       "\n",
       "               suv5         suv6         suv7        suv8       suv9  \\\n",
       "count  5.185000e+03  4817.000000  5185.000000  368.000000  368.00000   \n",
       "mean   3.012852e+07    30.460868     6.231823    1.426630   50.92663   \n",
       "min    3.092520e+05     1.000000     1.000000    1.000000    1.00000   \n",
       "25%    3.148632e+07    11.000000     2.000000    1.000000    2.00000   \n",
       "50%    3.148646e+07    20.000000     2.000000    1.000000   45.00000   \n",
       "75%    3.148655e+07    45.000000    13.000000    2.000000   94.00000   \n",
       "max    3.148667e+07   124.000000    33.000000    2.000000  113.00000   \n",
       "std    6.361543e+06    27.639562     6.889888    0.495261   39.38848   \n",
       "\n",
       "             suv10  ...          hv1                            hv2  \\\n",
       "count  5185.000000  ...  5185.000000                           5185   \n",
       "mean     16.007715  ...     8.344841  2022-01-26 12:35:41.253616384   \n",
       "min       1.000000  ...     2.000000            2021-01-01 00:00:00   \n",
       "25%       1.000000  ...     4.000000            2022-01-18 00:00:00   \n",
       "50%      12.000000  ...     4.000000            2022-02-12 00:00:00   \n",
       "75%      29.000000  ...    14.000000            2022-02-21 00:00:00   \n",
       "max     114.000000  ...    21.000000            2022-02-28 00:00:00   \n",
       "std      16.938089  ...     6.118074                            NaN   \n",
       "\n",
       "               hv3          hv4          hv5          hv8        hv10  \\\n",
       "count  5185.000000  5185.000000  5185.000000  5185.000000  240.000000   \n",
       "mean     53.827965     2.379942     1.903182     0.205786    1.966667   \n",
       "min       1.000000     1.000000     1.000000     0.000000    1.000000   \n",
       "25%      13.000000     2.000000     1.000000     0.000000    2.000000   \n",
       "50%      62.000000     3.000000     2.000000     0.000000    2.000000   \n",
       "75%      72.000000     3.000000     3.000000     0.000000    2.000000   \n",
       "max     161.000000     3.000000     3.000000     1.000000    2.000000   \n",
       "std      37.748223     0.734079     0.833264     0.404314    0.179881   \n",
       "\n",
       "                                 sv1          sv2          sv3  \n",
       "count                           5185  5185.000000  5185.000000  \n",
       "mean   2022-01-31 20:22:15.853423360    26.572420     0.213308  \n",
       "min              2021-01-01 00:00:00     1.000000     0.000000  \n",
       "25%              2022-01-31 00:00:00    12.000000     0.000000  \n",
       "50%              2022-02-17 00:00:00    15.000000     0.000000  \n",
       "75%              2022-02-23 00:00:00    23.000000     0.000000  \n",
       "max              2022-03-01 00:00:00    83.000000     1.000000  \n",
       "std                              NaN    26.466112     0.409683  \n",
       "\n",
       "[8 rows x 29 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Discreption about numetical columns\n",
    "df.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "26856e3b-6b94-4e5b-bb8d-394dad175bca",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 5185 entries, 0 to 5184\n",
      "Data columns (total 34 columns):\n",
      " #   Column  Non-Null Count  Dtype         \n",
      "---  ------  --------------  -----         \n",
      " 0   suv1    5185 non-null   int32         \n",
      " 1   suv2    5185 non-null   int32         \n",
      " 2   suv3    5185 non-null   int32         \n",
      " 3   suv4    5185 non-null   int32         \n",
      " 4   suv5    5185 non-null   int32         \n",
      " 5   suv6    4817 non-null   float64       \n",
      " 6   suv7    5185 non-null   int8          \n",
      " 7   suv8    368 non-null    float64       \n",
      " 8   suv9    368 non-null    float64       \n",
      " 9   suv10   5185 non-null   int16         \n",
      " 10  suv11   5169 non-null   float64       \n",
      " 11  suv12   4817 non-null   float64       \n",
      " 12  suv13   5185 non-null   object        \n",
      " 13  suv14   4805 non-null   float64       \n",
      " 14  pv1     5185 non-null   int8          \n",
      " 15  pv2     5185 non-null   int8          \n",
      " 16  pv3     5139 non-null   float64       \n",
      " 17  pv4     5139 non-null   float64       \n",
      " 18  pv5     5185 non-null   int8          \n",
      " 19  pv6     3690 non-null   float64       \n",
      " 20  hv1     5185 non-null   int8          \n",
      " 21  hv2     5185 non-null   datetime64[ns]\n",
      " 22  hv3     5185 non-null   int16         \n",
      " 23  hv4     5185 non-null   int8          \n",
      " 24  hv5     5185 non-null   int8          \n",
      " 25  hv6     5185 non-null   object        \n",
      " 26  hv7     5185 non-null   object        \n",
      " 27  hv8     5185 non-null   int8          \n",
      " 28  hv9     5185 non-null   object        \n",
      " 29  hv10    240 non-null    float64       \n",
      " 30  sv1     5185 non-null   datetime64[ns]\n",
      " 31  sv2     5185 non-null   int8          \n",
      " 32  sv3     5185 non-null   int8          \n",
      " 33  sv4     5185 non-null   object        \n",
      "dtypes: datetime64[ns](2), float64(10), int16(2), int32(5), int8(10), object(5)\n",
      "memory usage: 860.9+ KB\n"
     ]
    }
   ],
   "source": [
    "# Dataset information\n",
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5411f3f5-a858-4a24-b092-ec84392b2396",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "70c6615c-879e-4770-ad57-3d11f84158cb",
   "metadata": {},
   "source": [
    "- Column names are not properly representing the column data\n",
    "- Renaming of columns is necessory\n",
    "- Renaming is done with reference to metadata provided with dataset\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e182b2ec-0806-4fd9-a615-76c3e7f74068",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Renaming of columns\n",
    "df.rename(columns={'pv1': 'Gender', 'pv2': 'Age'}, inplace=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7921d422-fc31-45d0-a37e-c4dd6d2ca62c",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.rename(columns={'pv3': 'District', 'pv4': 'State', 'pv5': 'Country', 'pv6': 'location type',\n",
    "                  'hv1': 'Hospital ID', 'hv2': 'Date of Admission', 'hv3': 'Hospital department', 'hv4': 'Hospital Department type',\n",
    "                  'hv6': 'Clinical Diagnosis', 'hv7': 'Comorbidity', 'hv8': 'Whether device used?', 'hv9': 'Patient with Devices',\n",
    "                  'hv10': 'Prior Admission in Last 3 months', 'sv1': 'Date of sample Collection', 'sv2': 'Sample Type', \n",
    "                  'sv3': 'antibiotic prescription in last 1 months?', 'sv4': 'Antibiotics/Antifungals taken in last 1 month',\n",
    "                  'suv1': 'AMR ID'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "ad259c73-ba87-4d9a-9171-b777f6df015c",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.rename(columns={'hv5': 'Infection Type', 'suv2': 'Suscepbility ID', 'suv3': 'Sample ID', 'suv4': 'Hospital Patient Relation ID',\n",
    "                  'suv5': 'Patient ID', 'suv6': 'Antibiotic', 'suv7': 'Panel ID', 'suv8': 'special test type',\n",
    "                  'suv9': 'Special Test', 'suv10': 'Organism', 'suv11': 'Identification Method', 'suv12': 'Susceptibility Test Type',\n",
    "                  'suv13': 'Value', 'suv14': 'Remarks'}, inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f6cdb5dc-c098-46df-abb5-526d8297afa0",
   "metadata": {},
   "source": [
    "#### Replacing numerical values with original values in respective columns\n",
    "- In datadet all object data is replaced with numeric data\n",
    "- for my conveniance I am replacing numerical data with original string data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5d13780f-cd48-4be9-abb4-933c64de57f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 5185 entries, 0 to 5184\n",
      "Data columns (total 34 columns):\n",
      " #   Column                                         Non-Null Count  Dtype         \n",
      "---  ------                                         --------------  -----         \n",
      " 0   AMR ID                                         5185 non-null   int32         \n",
      " 1   Suscepbility ID                                5185 non-null   int32         \n",
      " 2   Sample ID                                      5185 non-null   int32         \n",
      " 3   Hospital Patient Relation ID                   5185 non-null   int32         \n",
      " 4   Patient ID                                     5185 non-null   int32         \n",
      " 5   Antibiotic                                     4817 non-null   float64       \n",
      " 6   Panel ID                                       5185 non-null   int8          \n",
      " 7   special test type                              368 non-null    float64       \n",
      " 8   Special Test                                   368 non-null    float64       \n",
      " 9   Organism                                       5185 non-null   int16         \n",
      " 10  Identification Method                          5169 non-null   float64       \n",
      " 11  Susceptibility Test Type                       4817 non-null   float64       \n",
      " 12  Value                                          5185 non-null   object        \n",
      " 13  Remarks                                        4805 non-null   float64       \n",
      " 14  Gender                                         5185 non-null   int8          \n",
      " 15  Age                                            5185 non-null   int8          \n",
      " 16  District                                       5139 non-null   float64       \n",
      " 17  State                                          5139 non-null   float64       \n",
      " 18  Country                                        5185 non-null   int8          \n",
      " 19  location type                                  3690 non-null   float64       \n",
      " 20  Hospital ID                                    5185 non-null   int8          \n",
      " 21  Date of Admission                              5185 non-null   datetime64[ns]\n",
      " 22  Hospital department                            5185 non-null   int16         \n",
      " 23  Hospital Department type                       5185 non-null   int8          \n",
      " 24  Infection Type                                 5185 non-null   int8          \n",
      " 25  Clinical Diagnosis                             5185 non-null   object        \n",
      " 26  Comorbidity                                    5185 non-null   object        \n",
      " 27  Whether device used?                           5185 non-null   int8          \n",
      " 28  Patient with Devices                           5185 non-null   object        \n",
      " 29  Prior Admission in Last 3 months               240 non-null    float64       \n",
      " 30  Date of sample Collection                      5185 non-null   datetime64[ns]\n",
      " 31  Sample Type                                    5185 non-null   int8          \n",
      " 32  antibiotic prescription in last 1 months?      5185 non-null   int8          \n",
      " 33  Antibiotics/Antifungals taken in last 1 month  5185 non-null   object        \n",
      "dtypes: datetime64[ns](2), float64(10), int16(2), int32(5), int8(10), object(5)\n",
      "memory usage: 860.9+ KB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c1bce73d-49af-490f-a91c-1f3b53149af8",
   "metadata": {},
   "outputs": [],
   "source": [
    "antibiotics = {1 :\"Amikacin\", 2: \"Amikacin\",3: \"Cefepime\",4: \"Cefepime\",5: \"Cefoperazone-sulbactam\",6: \"Cefoperazone-sulbactam\",7: \"Cefotaxime\",\n",
    "               8: \"Cefotaxime\",9: \"Ceftazidime\",10: \"Ceftazidime\",11: \"Ciprofloxacin\",12: \"Ciprofloxacin\",13: \"Colistin\",14: \"Doxycycline\",\n",
    "               15: \"Doxycycline\",16: \"Ertapenem\",17: \"Ertapenem\",18: \"Imipenem\",19: \"Imipenem\",20: \"Meropenem\",21: \"Meropenem\",22: \"Nitrofurantoin\",\n",
    "               23: \"Nitrofurantoin\",24: \"Norfloxacin\",25: \"Norfloxacin\",26: \"Piperacillin-tazobactam\",27: \"Piperacillin-tazobactam\",\n",
    "               28: \"Trimethoprim-sulfamethoxazole\",29: \"Trimethoprim-sulfamethoxazole\",30: \"Ampicillin\",31: \"Ampicillin\",32: \"Cefixime\",\n",
    "               33: \"Cefixime\",34: \"Ceftriaxone\",35: \"Ceftriaxone\",36: \"Chloramphenicol\",37: \"Chloramphenicol\",38: \"Ofloxacin\",39: \"Aztreonam\",\n",
    "               40: \"Aztreonam\",41: \"Colistin\",42: \"Colistin\",43: \"Gentamicin\",44: \"Gentamicin\",45: \"Levofloxacin\",46: \"Levofloxacin\",47:\"Netilmicin\",\n",
    "               48: \"Netilmicin\",49: \"Tobramycin\",50: \"Tobramycin\",52: \"Tetracycline\",53: \"Tetracycline\",58: \"Minocycline\",59: \"Minocycline\",\n",
    "               60: \"Ticarcillin-clavulanic acid\",62: \"Azithromycin\",63: \"Clindamycin\",64: \"Nalidixic acid\",65: \"Nalidixic acid\",66: \"Cefoxitin\",\n",
    "               67: \"Cefoxitin\",68: \"Clindamycin\",69: \"Clindamycin\",70: \"Erythromycin\",71: \"Erythromycin\",72: \"Linezolid\",73: \"Linezolid\",\n",
    "               74: \"Mupirocin High Level\",75: \"Mupirocin High Level\",76: \"Penicillin\",77: \"Penicillin\",78: \"Teicoplanin\",79: \"Teicoplanin\",\n",
    "               80: \"Tigecycline\",81: \"Vancomycin\",84: \"Vancomycin\",86: \"Gentamicin_HL\",87: \"Gentamicin_HL\",88: \"Anidulafungin\",89: \"Anidulafungin\",\n",
    "               90: \"Caspofungin\",91: \"Caspofungin\",92: \"Micafungin\",93: \"Micafungin\",94: \"Fluconazole\",95: \"Fluconazole\",96: \"Voriconazole\",\n",
    "               97: \"Voriconazole\",98: \"Amphotericin B\",99: \"Itraconazole\",100: \"Posaconazole\",101: \"Amphotericin B\",102: \"Itraconazole\",\n",
    "               103: \"Posaconazole\",104: \"Tigecycline\",105: \"Tigecycline\",106: \"Oxacillin\",107: \"Oxacillin\",108: \"Cefazolin\",109: \"Cefazolin\",\n",
    "               110: \"Fosfomycin\",111: \"Fosfomycin\",112: \"Azithromycin\",114: \"Pefloxacin\",115: \"Oritavancin\",116: \"Tedizolid\",117: \"Telavancin\",\n",
    "               118: \"Imipenem-relebactam\",119: \"Imipenem-relebactam\",120: \"Ceftazidime-avibactam\",121: \"Ceftazidime-avibactam\",\n",
    "               124: \"Meropenem-vaborbactam\",125: \"Meropenem-vaborbactam\",126: \"Polymixin B\",127: \"Daptomycin\",\n",
    "               128: \"Penicillin (Parental and Meningitis)\",129: \"Penicillin (Parental and Nonmeningitis)\",130: \"Penicillin (Oral)\",\n",
    "               131: \"Cefotaxime (Non-meningitis)\",132: \"Cefotaxime (Meningitis)\",133: \"Ceftriaxone (Non-meningitis)\",\n",
    "               134: \"Ceftriaxone (Meningitis)\",135: \"Ceftaroline\",136: \"Ceftaroline\" \n",
    "              }\n",
    "\n",
    "df['Antibiotic'] = df['Antibiotic'].replace(antibiotics)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d332bf5d-cd94-461f-881c-340dceb606cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "pannel = {1: \"Enterobacteriaceae-Urine\",2: \"Enterobacteriaceae-Non urine\",3: \"Typhoidal Salmonella\",4: \"Pseudomonas aeruginosa\",\n",
    "          5: \"Acinetobacter\",6: \"Burkholderia cepacia\",7: \"Stenotrophomonas maltophilia\",8: \"Shigella, DEC\",9: \"Faecal Salmonella, Arizona\",\n",
    "          10: \"Vibrio\",11: \"Campylobacter\",12: \"Aeromonas, Plesiomonas\",13: \"Staphylococcus aureus\",14: \"Staphylococcus Coagulase Negative\",\n",
    "          15: \"Enterococci faecium and spp - Urine\",17: \"Staphylococcus lugdunensis\",18: \"Staphylococcus pseudointermedius\",\n",
    "          19: \"Enterococci - Non urine\",20: \"Yeasts\",21: \"Candida glabrata\",22: \"Candida parapsilosis\",23: \"Candida krusei\",\n",
    "          24: \"Candida guilliermondii\",25: \"Aspergillus\",26: \"Aspergillus terreus\",27: \"Aspergillus nidulans\",28: \"Streptococcus pneumoniae\",\n",
    "          29: \"Streptococcus beta haemolytic group\",30: \"Streptococcus viridans\",31: \"Candida albicans\"\n",
    "         }\n",
    "\n",
    "df['Panel ID'] = df['Panel ID'].replace(pannel)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d68455ba-bcc3-46ab-9202-1ecc120eedb1",
   "metadata": {},
   "outputs": [],
   "source": [
    "special_test_type = {1: 'Genotypic',2: 'Phenotypic' }\n",
    "\n",
    "df['special test type'] = df['special test type'].replace(special_test_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "eb3a4c27-188f-4a1a-9e71-eaeab2557821",
   "metadata": {},
   "outputs": [],
   "source": [
    "special_test = {1: \"ESBL\",2: \"AmpC\",6: \"CMY\",7: \"DHA\",8: \"TEM\",10: \"Carba NP confirmatory test\",11: \"Carba NP screening test\",\n",
    "                12: \"Carbapenemase-Imipenem-Aminophenyl boronic acid\",13: \"Carbapenemase-Imipenem-Phenyl boronic acid\",\n",
    "                14: \"Carbapenemase-Imipenem-Cloxacillin\",15: \"Carbapenemase-Imipenem-EDTA\",\n",
    "                17: \"Efflux pump over expression-Levofloxacin-EPIPAbetaN\",18: \"CarbAcineto NP screening test\",\n",
    "                19: \"Outer membrane porin loss detection-OprD\",20: \"Outer membrane porin loss detection-CarO\",\n",
    "                21: \"Beta lactamase Penicillin zone edge test\",22: \"Beta lactamase Nitrocefin test\",23: \"PBP2a latex agglutination test\",\n",
    "                24: \"D-test for inducible Clindamycin resistance\",25: \"SHV\",26: \"OXA-1\",30: \"CTX-M-1\",31: \"CTX-M-2\",32: \"CTX-M-9\",34: \"CTX-M-8/25\",\n",
    "                35: \"ACC\",36: \"MOX\",38: \"CIT\",39: \"EBC\",40: \"VEB\",41: \"GES\",42: \"PER\",43: \"VIM\",44: \"IMP\",45: \"KPC\",50: \"NDM-1\",51: \"gyrA\",\n",
    "                52: \"gyrB\",53: \"parC\",54: \"parE\",55: \"qnrA\",56: \"qnrB\",57: \"qnrS\",58: \"acrR\",60: \"FOX\",61: \"SPM\",62: \"OXA 48 like\",63: \"MexAB-OprM\",\n",
    "                64: \"MexXY-OprM\",65: \"MexCD-OprJ\",66: \"MexEF-OprN\",67: \"OprD\",68: \"OXA 51\",69: \"OXA 23\",70: \"OXA 24\",71: \"OXA 58\",72: \"bla-OXA\",\n",
    "                73: \"dhfrIa\",74: \"sul1\",76: \"aac(6')-Ib-cr\",77: \"bla-TEM\",78: \"dfrA1\",79: \"sul2\",80: \"tetA\",81: \"nim\",83: \"tet\",84: \"tetO\",\n",
    "                85: \"aerA\",86: \"act\",87: \"hly\",88: \"mecA\",89: \"mupA\",90: \"mupB\",91: \"ermA\",92: \"ermB\",93: \"ermC\",94: \"vanA\",95: \"vanB\",96: \"vanC\",\n",
    "                97: \"vanHAX\",98: \"aac(6')-Ie\",99: \"aph(2'')-Ia\",100: \"aph(2'')-Ib\",101: \"aph(2'')-Ic\",102: \"aph(2'')-Id\",103: \"aph(3')-IIIa\",\n",
    "                104: \"ant(4')-Ia\",105: \"Carbapenemase Modified Hodge test\",106: \"alt\",107: \"AmpC\",108: \"CTX-M\",109: \"Modified Hodge test\",\n",
    "                110: \"Carba-NP test\",111: \"Modified carbapenem inactivation method (mCIM)\",112: \"Cfr\",113: \"GES\",114: \"ERG11-A114S\",\n",
    "                115: \"ERG11-Y132F/H\",116: \"ERG11-K143R/Q\",117: \"ERG11-D446E\",118: \"ERG11-G448E\",119: \"ERG11-G450E\",120: \"ERG11- G464S\",\n",
    "                121: \"ERG11-R467K\",122: \"FKS1-S645P/F/Y\",123: \"FKS1-F641S/L\",124: \"FKS1-D648Y\",125: \"FKS1-P649H\",126: \"FKS1-R1361H\",\n",
    "                127: \"Zygosity of the mutation\",128: \"Any other genetic alteration\",129: \"ERG11-Y132F\",130: \"ERG11-G464S\",131: \"ERG11-Y257H\",\n",
    "                132: \"ERG11-V125A\",133: \"FKS1-S654P\",135: \"ERG11-K143R\",136: \"FKS1-S639P\",137: \"FKS1-S639F\",139: \"FKS1-F635Y\",140: \"FKS1-F635L\",\n",
    "                141: \"FKS1-R1354S\",142: \"FKS1-S663F/P\",143: \"FKS1-S659S/V/Y/ÃƒÂ¢Ã‹â€ Ã¢â‚¬ \",144: \"FKS1- L664R\",145: \"FKS1-P667T\",146: \"FKS2-S629P\",\n",
    "                147: \"FKS2-F625S\",148: \"FKS2-F623E/G\",149: \"TR34/L98H\",150: \"TR46/Y121F/T289A\",151: \"Cyp51A-G54\",152: \"Cyp51A-M220\",\n",
    "                153: \"Cyp51A-Y319H\",154: \"Cyp51A-S196F\",155: \"Cyp51A-N423D\",\n",
    "               }\n",
    "\n",
    "df['Special Test'] = df['Special Test'].replace(special_test)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f97ecddd-0ae1-4911-aac8-dc2412cec1cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "text = '''1 \"Escherichia coli\"\n",
    "2 \"Citrobacter freundii\"\n",
    "3 \"Citrobacter koseri\"\n",
    "4 \"Citrobacter spp.\"\n",
    "5 \"Klebsiella pneumoniae\"\n",
    "6 \"Klebsiella oxytoca \"\n",
    "7 \"Klebsiella spp. \"\n",
    "8 \"Klebsiella (Enterobacter) aerogenes\"\n",
    "9 \"Enterobacter cloacae \"\n",
    "10 \"Enterobacter spp.\"\n",
    "11 \"Hafnia alvei \"\n",
    "12 \"Serratia marcescens \"\n",
    "13 \"Proteus mirabilis \"\n",
    "14 \"Proteus vulgaris \"\n",
    "15 \"Morganella morganii \"\n",
    "16 \"Providencia rettgeri \"\n",
    "17 \"Providencia stuartii \"\n",
    "18 \"Salmonella Typhi \"\n",
    "19 \"Salmonella Paratyphi A \"\n",
    "20 \"Salmonella spp. \"\n",
    "21 \"Pseudomonas aeruginosa\"\n",
    "23 \"Acinetobacter baumannii \"\n",
    "24 \"Acinetobacter calcoaceticus \"\n",
    "25 \"Acinetobacter lwoffii \"\n",
    "26 \"Acinetobacter spp. \"\n",
    "27 \"Stenotrophomonas maltophilia \"\n",
    "28 \"Burkholderia cepacia \"\n",
    "29 \"Staphylococcus aureus\"\n",
    "32 \"Staphylococcus lugdunensis\"\n",
    "35 \"Staphylococcus haemolyticus\"\n",
    "36 \"Staphylococcus epidermidis\"\n",
    "37 \"Staphylococcus hominis\"\n",
    "38 \"Staphylococcus saprophyticus\"\n",
    "39 \"Enterococcus faecalis\"\n",
    "40 \"Enterococcus faecium\"\n",
    "41 \"Enterococcus spp.\"\n",
    "42 \"Aeromonas spp. \"\n",
    "43 \"Arizona spp. \"\n",
    "44 \"Vibrio cholerae \"\n",
    "45 \"Vibrio parahaemolyticus \"\n",
    "46 \"Vibrio spp.\"\n",
    "47 \"Escherichia coli Diarrhoeagenic\"\n",
    "48 \"Campylobacter jejuni \"\n",
    "50 \"Shigella dysenteriae\"\n",
    "51 \"Shigella flexneri \"\n",
    "52 \"Shigella boydii\"\n",
    "53 \"Shigella sonnei \"\n",
    "54 \"Shigella spp. \"\n",
    "55 \"Salmonella spp. Faecal\"\n",
    "56 \"Salmonella Typhimurium Faecal\"\n",
    "57 \"Salmonella Enteritidis \"\n",
    "58 \"Salmonella Newport \"\n",
    "59 \"Salmonella Heidelberg \"\n",
    "60 \"Plesiomonas shigelloides \"\n",
    "62 \"Staphylococcus spp.\"\n",
    "63 \"Staphylococcus pseudointermedius\"\n",
    "64 \"Aspergillus flavus \"\n",
    "65 \"Aspergillus fumigatus \"\n",
    "66 \"Aspergillus terreus \"\n",
    "67 \"Aspergillus versicolor \"\n",
    "68 \"Aspergillus nidulans \"\n",
    "69 \"Aspergillus niger \"\n",
    "70 \"Blastobortrys adeninivorans\"\n",
    "71 \"Candida haemulonii\"\n",
    "72 \"Candida krusei\"\n",
    "73 \"Candida auris\"\n",
    "74 \"Candida vishwanathii\"\n",
    "75 \"Candida catenulata\"\n",
    "76 \"Candida chiropterorum\"\n",
    "77 \"Candida famata\"\n",
    "78 \"Candida intermedia\"\n",
    "79 \"Candida kefyr\"\n",
    "80 \"Candida lipolytica\"\n",
    "81 \"Candida lusitaniae\"\n",
    "82 \"Candida nivariensis\"\n",
    "83 \"Candida pararugosa\"\n",
    "84 \"Candida pelliculosa\"\n",
    "85 \"Candida rugosa\"\n",
    "86 \"Candida utilis\"\n",
    "87 \"Candida albicans \"\n",
    "88 \"Candida guilliermondii \"\n",
    "89 \"Candida glabrata \"\n",
    "90 \"Candida parapsilosis \"\n",
    "91 \"Candida tropicalis\"\n",
    "92 \"Cryptococcus neoformans\"\n",
    "93 \"Cryptococcus neoformans var gattii\"\n",
    "94 \"Cryptococcus neoformans var grubii\"\n",
    "95 \"Debaromyces hansenii\"\n",
    "96 \"Kodamaea ohmeri\"\n",
    "97 \"Pichia fabianii\"\n",
    "98 \"Pichia jadinii\"\n",
    "99 \"Pichia kudriavzevii\"\n",
    "100 \"Pseudozyma aphidis\"\n",
    "101 \"Pseudozyma churashimaensis\"\n",
    "102 \"Rhizopus oryzae\"\n",
    "103 \"Rhizopus homothallicus\"\n",
    "104 \"Rhizopus microsporus\"\n",
    "105 \"Rhodotorula mucilaginosa\"\n",
    "106 \"Rhodotorula glutinis\"\n",
    "107 \"Saccharomyces cerevisiae\"\n",
    "108 \"Trichosporon asahii\"\n",
    "109 \"Trichosporon mucoides\"\n",
    "110 \"Trichophyton mentagrophytes\"\n",
    "111 \"Acinetobacter baumanii-calcoaceticus complex\"\n",
    "112 \"Burkholderia cepacia complex\"\n",
    "113 \"Streptococcus pneumoniae\"\n",
    "114 \"Streptococcus pyogenes\"\n",
    "115 \"Streptococcus agalactiae\"\n",
    "116 \"Streptococcus viridans\"\n",
    "117 \"Wickeromyces anomalus\"\n",
    "118 \"Rhizopus arrhizus\"\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "8ab7f721-ba13-4f5a-94f1-8c25161a12c4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{1: 'Escherichia coli', 2: 'Citrobacter freundii', 3: 'Citrobacter koseri', 4: 'Citrobacter spp.', 5: 'Klebsiella pneumoniae', 6: 'Klebsiella oxytoca ', 7: 'Klebsiella spp. ', 8: 'Klebsiella (Enterobacter) aerogenes', 9: 'Enterobacter cloacae ', 10: 'Enterobacter spp.', 11: 'Hafnia alvei ', 12: 'Serratia marcescens ', 13: 'Proteus mirabilis ', 14: 'Proteus vulgaris ', 15: 'Morganella morganii ', 16: 'Providencia rettgeri ', 17: 'Providencia stuartii ', 18: 'Salmonella Typhi ', 19: 'Salmonella Paratyphi A ', 20: 'Salmonella spp. ', 21: 'Pseudomonas aeruginosa', 23: 'Acinetobacter baumannii ', 24: 'Acinetobacter calcoaceticus ', 25: 'Acinetobacter lwoffii ', 26: 'Acinetobacter spp. ', 27: 'Stenotrophomonas maltophilia ', 28: 'Burkholderia cepacia ', 29: 'Staphylococcus aureus', 32: 'Staphylococcus lugdunensis', 35: 'Staphylococcus haemolyticus', 36: 'Staphylococcus epidermidis', 37: 'Staphylococcus hominis', 38: 'Staphylococcus saprophyticus', 39: 'Enterococcus faecalis', 40: 'Enterococcus faecium', 41: 'Enterococcus spp.', 42: 'Aeromonas spp. ', 43: 'Arizona spp. ', 44: 'Vibrio cholerae ', 45: 'Vibrio parahaemolyticus ', 46: 'Vibrio spp.', 47: 'Escherichia coli Diarrhoeagenic', 48: 'Campylobacter jejuni ', 50: 'Shigella dysenteriae', 51: 'Shigella flexneri ', 52: 'Shigella boydii', 53: 'Shigella sonnei ', 54: 'Shigella spp. ', 55: 'Salmonella spp. Faecal', 56: 'Salmonella Typhimurium Faecal', 57: 'Salmonella Enteritidis ', 58: 'Salmonella Newport ', 59: 'Salmonella Heidelberg ', 60: 'Plesiomonas shigelloides ', 62: 'Staphylococcus spp.', 63: 'Staphylococcus pseudointermedius', 64: 'Aspergillus flavus ', 65: 'Aspergillus fumigatus ', 66: 'Aspergillus terreus ', 67: 'Aspergillus versicolor ', 68: 'Aspergillus nidulans ', 69: 'Aspergillus niger ', 70: 'Blastobortrys adeninivorans', 71: 'Candida haemulonii', 72: 'Candida krusei', 73: 'Candida auris', 74: 'Candida vishwanathii', 75: 'Candida catenulata', 76: 'Candida chiropterorum', 77: 'Candida famata', 78: 'Candida intermedia', 79: 'Candida kefyr', 80: 'Candida lipolytica', 81: 'Candida lusitaniae', 82: 'Candida nivariensis', 83: 'Candida pararugosa', 84: 'Candida pelliculosa', 85: 'Candida rugosa', 86: 'Candida utilis', 87: 'Candida albicans ', 88: 'Candida guilliermondii ', 89: 'Candida glabrata ', 90: 'Candida parapsilosis ', 91: 'Candida tropicalis', 92: 'Cryptococcus neoformans', 93: 'Cryptococcus neoformans var gattii', 94: 'Cryptococcus neoformans var grubii', 95: 'Debaromyces hansenii', 96: 'Kodamaea ohmeri', 97: 'Pichia fabianii', 98: 'Pichia jadinii', 99: 'Pichia kudriavzevii', 100: 'Pseudozyma aphidis', 101: 'Pseudozyma churashimaensis', 102: 'Rhizopus oryzae', 103: 'Rhizopus homothallicus', 104: 'Rhizopus microsporus', 105: 'Rhodotorula mucilaginosa', 106: 'Rhodotorula glutinis', 107: 'Saccharomyces cerevisiae', 108: 'Trichosporon asahii', 109: 'Trichosporon mucoides', 110: 'Trichophyton mentagrophytes', 111: 'Acinetobacter baumanii-calcoaceticus complex', 112: 'Burkholderia cepacia complex', 113: 'Streptococcus pneumoniae', 114: 'Streptococcus pyogenes', 115: 'Streptococcus agalactiae', 116: 'Streptococcus viridans', 117: 'Wickeromyces anomalus', 118: 'Rhizopus arrhizus'}\n"
     ]
    }
   ],
   "source": [
    "species_dict = {}\n",
    "\n",
    "for line in text.strip().split(\"\\n\"):\n",
    "    parts = line.split(\" \", 1)  # Split on the first space\n",
    "    key = int(parts[0])  # Convert the first part to an integer\n",
    "    value = parts[1].strip('\"')  # Remove surrounding quotes\n",
    "    species_dict[key] = value  # Add to dictionary\n",
    "\n",
    "# Print dictionary\n",
    "print(species_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "4d1de1e7-31d2-4bfe-b686-b801e37b0a1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "df['Organism'] = df['Organism'].replace(species_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "9dae982f-e6b4-42bf-bf15-8f59b15da6fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "text2 = '''1 \"Biochemical test\"\n",
    "2 \"Proteomics\"\n",
    "3 \"Vitek-2\"\n",
    "4 \"MaldiToff\"\n",
    "5 \"16s rRNA Sequencing\"\n",
    "6 \"Whole Genome Sequencing\"\n",
    "7 \"Conventional identification by microscopy/phenotype\"\n",
    "'''\n",
    "identification = {}\n",
    "\n",
    "for line in text2.strip().split(\"\\n\"):\n",
    "    parts = line.split(\" \", 1) \n",
    "    key = int(parts[0])\n",
    "    value = parts[1].strip('\"')\n",
    "    identification[key] = value\n",
    "\n",
    "df['Identification Method'] = df['Identification Method'].replace(species_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "c2815885-9e62-46fd-8455-59876db64d53",
   "metadata": {},
   "outputs": [],
   "source": [
    "text3 = '''1 \"AUTOMATED\"\n",
    "2 \"DISK\"\n",
    "3 \"ETEST\"\n",
    "4 \"MIC\"\n",
    "'''\n",
    "Susceptibility_Test_Type = {}\n",
    "\n",
    "for line in text3.strip().split(\"\\n\"):\n",
    "    parts = line.split(\" \", 1) \n",
    "    key = int(parts[0])\n",
    "    value = parts[1].strip('\"')\n",
    "    Susceptibility_Test_Type[key] = value\n",
    "\n",
    "df['Susceptibility Test Type'] = df['Susceptibility Test Type'].replace(Susceptibility_Test_Type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "ca45097d-0b08-421f-ad21-e68ecced5d09",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 5185 entries, 0 to 5184\n",
      "Data columns (total 34 columns):\n",
      " #   Column                                         Non-Null Count  Dtype         \n",
      "---  ------                                         --------------  -----         \n",
      " 0   AMR ID                                         5185 non-null   int32         \n",
      " 1   Suscepbility ID                                5185 non-null   int32         \n",
      " 2   Sample ID                                      5185 non-null   int32         \n",
      " 3   Hospital Patient Relation ID                   5185 non-null   int32         \n",
      " 4   Patient ID                                     5185 non-null   int32         \n",
      " 5   Antibiotic                                     4817 non-null   object        \n",
      " 6   Panel ID                                       5185 non-null   object        \n",
      " 7   special test type                              368 non-null    object        \n",
      " 8   Special Test                                   368 non-null    object        \n",
      " 9   Organism                                       5185 non-null   object        \n",
      " 10  Identification Method                          5169 non-null   object        \n",
      " 11  Susceptibility Test Type                       4817 non-null   object        \n",
      " 12  Value                                          5185 non-null   object        \n",
      " 13  Remarks                                        4805 non-null   float64       \n",
      " 14  Gender                                         5185 non-null   int8          \n",
      " 15  Age                                            5185 non-null   int8          \n",
      " 16  District                                       5139 non-null   float64       \n",
      " 17  State                                          5139 non-null   float64       \n",
      " 18  Country                                        5185 non-null   int8          \n",
      " 19  location type                                  3690 non-null   float64       \n",
      " 20  Hospital ID                                    5185 non-null   int8          \n",
      " 21  Date of Admission                              5185 non-null   datetime64[ns]\n",
      " 22  Hospital department                            5185 non-null   int16         \n",
      " 23  Hospital Department type                       5185 non-null   int8          \n",
      " 24  Infection Type                                 5185 non-null   int8          \n",
      " 25  Clinical Diagnosis                             5185 non-null   object        \n",
      " 26  Comorbidity                                    5185 non-null   object        \n",
      " 27  Whether device used?                           5185 non-null   int8          \n",
      " 28  Patient with Devices                           5185 non-null   object        \n",
      " 29  Prior Admission in Last 3 months               240 non-null    float64       \n",
      " 30  Date of sample Collection                      5185 non-null   datetime64[ns]\n",
      " 31  Sample Type                                    5185 non-null   int8          \n",
      " 32  antibiotic prescription in last 1 months?      5185 non-null   int8          \n",
      " 33  Antibiotics/Antifungals taken in last 1 month  5185 non-null   object        \n",
      "dtypes: datetime64[ns](2), float64(5), int16(1), int32(5), int8(9), object(12)\n",
      "memory usage: 926.7+ KB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "670ef628-cbe2-4d49-a6ff-c1c16a03cf04",
   "metadata": {},
   "outputs": [],
   "source": [
    "text4 = '''1 \"susceptible \"\n",
    "2 \"Intermediate\"\n",
    "3 \"resistant\" \n",
    "'''\n",
    "Remarks_ = {}\n",
    "\n",
    "for line in text4.strip().split(\"\\n\"):\n",
    "    parts = line.split(\" \", 1) \n",
    "    key = int(parts[0])\n",
    "    value = parts[1].strip('\"')\n",
    "    Remarks_[key] = value\n",
    "\n",
    "df['Remarks'] = df['Remarks'].replace(Remarks_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "e7c7c812-a47a-4851-bd84-87cee869c8e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "text5 = '''1 \"female\"\n",
    "2 â€œmale\"\n",
    "3 \"transgender\"\n",
    " '''\n",
    "\n",
    "sex = {}\n",
    "\n",
    "for line in text5.strip().split(\"\\n\"):\n",
    "    parts = line.split(\" \", 1) \n",
    "    key = int(parts[0])\n",
    "    value = parts[1].strip('\"')\n",
    "    sex[key] = value\n",
    "\n",
    "df['Gender'] = df['Gender'].replace(sex)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "d69fa50b-56ad-422b-b896-0592b589501b",
   "metadata": {},
   "outputs": [],
   "source": [
    "text6 = '''1 \"Nicobar\"\n",
    "2 \"North and Middle Andaman\"\n",
    "3 \"South Andaman\"\n",
    "4 \"Anantapur\"\n",
    "5 \"Chittoor\"\n",
    "6 \"East Godavari\"\n",
    "7 \"Guntur\"\n",
    "8 \"Kadapa\"\n",
    "9 \"Krishna\"\n",
    "10 \"Kurnool\"\n",
    "11 \"Nellore\"\n",
    "12 \"Prakasam\"\n",
    "13 \"Srikakulam\"\n",
    "14 \"Visakhapatnam\"\n",
    "15 \"Vizianagaram\"\n",
    "16 \"West Godavari\"\n",
    "17 \"Anjaw\"\n",
    "18 \"Changlang\"\n",
    "19 \"Dibang Valley\"\n",
    "20 \"East Kameng\"\n",
    "21 \"East Siang\"\n",
    "22 \"Kurung Kumey\"\n",
    "23 \"Lohit\"\n",
    "24 \"Longding\"\n",
    "25 \"Lower Dibang Valley\"\n",
    "26 \"Lower Subansiri\"\n",
    "27 \"Papum Pare\"\n",
    "28 \"Tawang\"\n",
    "29 \"Tirap\"\n",
    "30 \"Upper Siang\"\n",
    "31 \"Upper Subansiri\"\n",
    "32 \"West Kameng\"\n",
    "33 \"West Siang\"\n",
    "34 \"Baksa\"\n",
    "35 \"Barpeta\"\n",
    "36 \"Bongaigaon\"\n",
    "37 \"Cachar\"\n",
    "38 \"Chirang\"\n",
    "39 \"Darrang\"\n",
    "40 \"Dhemaji\"\n",
    "41 \"Dhubri\"\n",
    "42 \"Dibrugarh\"\n",
    "43 \"Dima Hasao\"\n",
    "44 \"Goalpara\"\n",
    "45 \"Golaghat\"\n",
    "46 \"Hailakandi\"\n",
    "47 \"Jorhat\"\n",
    "48 \"Kamrup\"\n",
    "49 \"Kamrup Metropolitan\"\n",
    "50 \"Karbi Anglong\"\n",
    "51 \"Karimganj\"\n",
    "52 \"Kokrajhar\"\n",
    "53 \"Lakhimpur\"\n",
    "54 \"Morigaon\"\n",
    "55 \"Nagaon\"\n",
    "56 \"Nalbari\"\n",
    "57 \"Sivasagar\"\n",
    "58 \"Sonitpur\"\n",
    "59 \"Tinsukia\"\n",
    "60 \"Udalguri\"\n",
    "61 \"Araria\"\n",
    "62 \"Arwal\"\n",
    "63 \"Aurangabad\"\n",
    "64 \"Banka\"\n",
    "65 \"Begusarai\"\n",
    "66 \"Bhagalpur\"\n",
    "67 \"Bhojpur\"\n",
    "68 \"Buxar\"\n",
    "69 \"Darbhanga\"\n",
    "70 \"East Champaran\"\n",
    "71 \"Gaya\"\n",
    "72 \"Gopalganj\"\n",
    "73 \"Jamui\"\n",
    "74 \"Jehanabad\"\n",
    "75 \"Kaimur\"\n",
    "76 \"Katihar\"\n",
    "77 \"Khagaria\"\n",
    "78 \"Kishanganj\"\n",
    "79 \"Lakhisarai\"\n",
    "80 \"Madhepura\"\n",
    "81 \"Madhubani\"\n",
    "82 \"Munger\"\n",
    "83 \"Muzaffarpur\"\n",
    "84 \"Nalanda\"\n",
    "85 \"Nawada\"\n",
    "86 \"Patna\"\n",
    "87 \"Purnia\"\n",
    "88 \"Rohtas\"\n",
    "89 \"Saharsa\"\n",
    "90 \"Samastipur\"\n",
    "91 \"Saran\"\n",
    "92 \"Sheikhpura\"\n",
    "93 \"Sheohar\"\n",
    "94 \"Sitamarhi\"\n",
    "95 \"Siwan\"\n",
    "96 \"Supaul\"\n",
    "97 \"Vaishali\"\n",
    "98 \"West Champaran\"\n",
    "99 \"Chandigarh\"\n",
    "100 \"Balrampur\"\n",
    "101 \"Bastar\"\n",
    "102 \"Bemetara\"\n",
    "103 \"Bijapur\"\n",
    "104 \"Bilaspur\"\n",
    "105 \"Dantewada\"\n",
    "106 \"Dhamtari\"\n",
    "107 \"Durg\"\n",
    "108 \"Gariaband\"\n",
    "109 \"Janjgir-Champa\"\n",
    "110 \"Jashpur\"\n",
    "111 \"Kanker\"\n",
    "112 \"Kawardha\"\n",
    "113 \"Kondagaon\"\n",
    "114 \"Korba\"\n",
    "115 \"Koriya\"\n",
    "116 \"Mahasamund\"\n",
    "117 \"Mungeli\"\n",
    "118 \"Narayanpur\"\n",
    "119 \"Raigarh\"\n",
    "120 \"Raipur\"\n",
    "121 \"Rajnandgaon\"\n",
    "122 \"Sukma\"\n",
    "123 \"Surajpur\"\n",
    "124 \"Surguja\"\n",
    "125 \"Dadra and Nagar Haveli\"\n",
    "126 \"Daman\"\n",
    "127 \"Diu\"\n",
    "128 \"Central Delhi\"\n",
    "129 \"East Delhi\"\n",
    "130 \"New Delhi\"\n",
    "131 \"North Delhi\"\n",
    "132 \"North East Delhi\"\n",
    "133 \"North West Delhi\"\n",
    "134 \"Shahdara\"\n",
    "135 \"South Delhi\"\n",
    "136 \"South East Delhi\"\n",
    "137 \"South West Delhi\"\n",
    "138 \"West Delhi\"\n",
    "139 \"Margao\"\n",
    "140 \"Panji\"\n",
    "141 \"Ahmedabad\"\n",
    "142 \"Amreli district\"\n",
    "143 \"Anand\"\n",
    "144 \"Aravalli\"\n",
    "145 \"Banaskantha\"\n",
    "146 \"Bharuch\"\n",
    "147 \"Bhavnagar\"\n",
    "148 \"Botad\"\n",
    "149 \"Chhota Udaipur\"\n",
    "150 \"Dahod\"\n",
    "151 \"Dang\"\n",
    "152 \"Devbhoomi Dwarka\"\n",
    "153 \"Gandhinagar\"\n",
    "154 \"Gir Somnath\"\n",
    "155 \"Jamnagar\"\n",
    "156 \"Junagadh\"\n",
    "157 \"Kheda\"\n",
    "158 \"Kutch\"\n",
    "159 \"Mahisagar\"\n",
    "160 \"Mehsana\"\n",
    "161 \"Morbi\"\n",
    "162 \"Narmada\"\n",
    "163 \"Navsari\"\n",
    "164 \"Panchmahal\"\n",
    "165 \"Patan\"\n",
    "166 \"Porbandar\"\n",
    "167 \"Rajkot\"\n",
    "168 \"Sabarkantha\"\n",
    "169 \"Surat\"\n",
    "170 \"Surendranagar\"\n",
    "171 \"Tapi\"\n",
    "172 \"Vadodara\"\n",
    "173 \"Valsad\"\n",
    "174 \"Ambala\"\n",
    "175 \"Bhiwani\"\n",
    "176 \"Faridabad\"\n",
    "177 \"Fatehabad\"\n",
    "178 \"Gurgaon\"\n",
    "179 \"Hissar\"\n",
    "180 \"Jhajjar\"\n",
    "181 \"Jind\"\n",
    "182 \"Kaithal\"\n",
    "183 \"Karnal\"\n",
    "184 \"Kurukshetra\"\n",
    "185 \"Mahendragarh\"\n",
    "186 \"Mewat\"\n",
    "187 \"Palwal\"\n",
    "188 \"Panchkula\"\n",
    "189 \"Panipat\"\n",
    "190 \"Rewari\"\n",
    "191 \"Rohtak\"\n",
    "192 \"Sirsa\"\n",
    "193 \"Sonipat\"\n",
    "194 \"Yamuna Nagar\"\n",
    "195 \"Bilaspur\"\n",
    "196 \"Chamba\"\n",
    "197 \"Hamirpur\"\n",
    "198 \"Kangra\"\n",
    "199 \"Kinnaur\"\n",
    "200 \"Kullu\"\n",
    "201 \"Lahaul and Spiti\"\n",
    "202 \"Mandi\"\n",
    "203 \"Shimla\"\n",
    "204 \"Sirmaur\"\n",
    "205 \"Solan\"\n",
    "206 \"Una\"\n",
    "207 \"Anantnag\"\n",
    "208 \"Badgam\"\n",
    "209 \"Bandipora\"\n",
    "210 \"Baramulla\"\n",
    "211 \"Doda\"\n",
    "212 \"Ganderbal\"\n",
    "213 \"Jammu\"\n",
    "214 \"Kargil\"\n",
    "215 \"Kathua\"\n",
    "216 \"Kishtwar\"\n",
    "217 \"Kulgam\"\n",
    "218 \"Kupwara\"\n",
    "219 \"Leh\"\n",
    "220 \"Poonch\"\n",
    "221 \"Pulwama\"\n",
    "222 \"Rajouri\"\n",
    "223 \"Ramban\"\n",
    "224 \"Reasi\"\n",
    "225 \"Samba\"\n",
    "226 \"Shopian\"\n",
    "227 \"Srinagar\"\n",
    "228 \"Udhampur\"\n",
    "229 \"Bokaro\"\n",
    "230 \"Chatra\"\n",
    "231 \"Deoghar\"\n",
    "232 \"Dhanbad\"\n",
    "233 \"Dumka\"\n",
    "234 \"East Singhbhum\"\n",
    "235 \"Garhwa\"\n",
    "236 \"Giridih\"\n",
    "237 \"Godda\"\n",
    "238 \"Gumla\"\n",
    "239 \"Hazaribag\"\n",
    "240 \"Jamtara\"\n",
    "241 \"Khunti\"\n",
    "242 \"Koderma\"\n",
    "243 \"Latehar\"\n",
    "244 \"Lohardaga\"\n",
    "245 \"Pakur\"\n",
    "246 \"Palamu\"\n",
    "247 \"Ramgarh\"\n",
    "248 \"Ranchi\"\n",
    "249 \"Sahibganj\"\n",
    "250 \"Seraikela Kharsawan\"\n",
    "251 \"Simdega\"\n",
    "252 \"West Singhbhum\"\n",
    "253 \"Bagalkot\"\n",
    "254 \"Bangalore Rural\"\n",
    "255 \"Bangalore Urban\"\n",
    "256 \"Belgaum\"\n",
    "257 \"Bellary\"\n",
    "258 \"Bidar\"\n",
    "259 \"Chamarajnagar\"\n",
    "260 \"Chikkaballapur\"\n",
    "261 \"Chikkamagaluru\"\n",
    "262 \"Chitradurga\"\n",
    "263 \"Dakshina Kannada\"\n",
    "264 \"Davanagere\"\n",
    "265 \"Dharwad\"\n",
    "266 \"Gadag\"\n",
    "267 \"Gulbarga\"\n",
    "268 \"Hassan\"\n",
    "269 \"Haveri district\"\n",
    "270 \"Kodagu\"\n",
    "271 \"Kolar\"\n",
    "272 \"Koppal\"\n",
    "273 \"Mandya\"\n",
    "274 \"Mysore\"\n",
    "275 \"Raichur\"\n",
    "276 \"Ramanagara\"\n",
    "277 \"Shimoga\"\n",
    "278 \"Tumkur\"\n",
    "279 \"Udupi\"\n",
    "280 \"Uttara Kannada\"\n",
    "281 \"Vijayapura\"\n",
    "282 \"Yadgir\"\n",
    "283 \"Alappuzha\"\n",
    "284 \"Ernakulam\"\n",
    "285 \"Idukki\"\n",
    "286 \"Kannur\"\n",
    "287 \"Kasaragod\"\n",
    "288 \"Kollam\"\n",
    "289 \"Kottayam\"\n",
    "290 \"Kozhikode\"\n",
    "291 \"Malappuram\"\n",
    "292 \"Palakkad\"\n",
    "293 \"Pathanamthitta\"\n",
    "294 \"Thiruvananthapuram\"\n",
    "295 \"Thrissur\"\n",
    "296 \"Wayanad\"\n",
    "297 \"Agar Malwa\"\n",
    "298 \"Alirajpur\"\n",
    "299 \"Anuppur\"\n",
    "300 \"Ashok Nagar\"\n",
    "301 \"Balaghat\"\n",
    "302 \"Barwani\"\n",
    "303 \"Betul\"\n",
    "304 \"Bhind\"\n",
    "305 \"Bhopal\"\n",
    "306 \"Burhanpur\"\n",
    "307 \"Chhatarpur\"\n",
    "308 \"Chhindwara\"\n",
    "309 \"Damoh\"\n",
    "310 \"Datia\"\n",
    "311 \"Dewas\"\n",
    "312 \"Dhar\"\n",
    "313 \"Dindori\"\n",
    "314 \"Guna\"\n",
    "315 \"Gwalior\"\n",
    "316 \"Harda\"\n",
    "317 \"Hoshangabad\"\n",
    "318 \"Indore\"\n",
    "319 \"Jabalpur\"\n",
    "320 \"Jhabua\"\n",
    "321 \"Katni\"\n",
    "322 \"Khandwa (East Nimar)\"\n",
    "323 \"Khargone (West Nimar)\"\n",
    "324 \"Mandla\"\n",
    "325 \"Mandsaur\"\n",
    "326 \"Morena\"\n",
    "327 \"Narsinghpur\"\n",
    "328 \"Neemuch\"\n",
    "329 \"Panna\"\n",
    "330 \"Raisen\"\n",
    "331 \"Rajgarh\"\n",
    "332 \"Ratlam\"\n",
    "333 \"Rewa\"\n",
    "334 \"Sagar\"\n",
    "335 \"Satna\"\n",
    "336 \"Sehore\"\n",
    "337 \"Seoni\"\n",
    "338 \"Shahdol\"\n",
    "339 \"Shajapur\"\n",
    "340 \"Sheopur\"\n",
    "341 \"Shivpuri\"\n",
    "342 \"Sidhi\"\n",
    "343 \"Singrauli\"\n",
    "344 \"Tikamgarh\"\n",
    "345 \"Ujjain\"\n",
    "346 \"Umaria\"\n",
    "347 \"Vidisha\"\n",
    "348 \"Agar Malwa\"\n",
    "349 \"Ahmednagar\"\n",
    "350 \"Akola\"\n",
    "351 \"Aurangabad\"\n",
    "352 \"Beed\"\n",
    "353 \"Bhandara\"\n",
    "354 \"Buldhana\"\n",
    "355 \"Chandrapur\"\n",
    "356 \"Dhule\"\n",
    "357 \"Gadchiroli\"\n",
    "358 \"Gondia\"\n",
    "359 \"Hingoli\"\n",
    "360 \"Jalgaon\"\n",
    "361 \"Jalna\"\n",
    "362 \"Kolhapur\"\n",
    "363 \"Latur\"\n",
    "364 \"Mumbai City\"\n",
    "365 \"Mumbai suburban\"\n",
    "366 \"Nagpur\"\n",
    "367 \"Nanded\"\n",
    "368 \"Nandurbar\"\n",
    "369 \"Nashik\"\n",
    "370 \"Osmanabad\"\n",
    "371 \"Palghar\"\n",
    "372 \"Parbhani\"\n",
    "373 \"Pune\"\n",
    "374 \"Raigad\"\n",
    "375 \"Ratnagiri\"\n",
    "376 \"Sangli\"\n",
    "377 \"Satara\"\n",
    "378 \"Sindhudurg\"\n",
    "379 \"Solapur\"\n",
    "380 \"Thane\"\n",
    "381 \"Wardha\"\n",
    "382 \"Washim\"\n",
    "383 \"Yavatmal\"\n",
    "384 \"Bishnupur\"\n",
    "385 \"Chandel\"\n",
    "386 \"Churachandpur\"\n",
    "387 \"Imphal East\"\n",
    "388 \"Imphal West\"\n",
    "389 \"Senapati\"\n",
    "390 \"Tamenglong\"\n",
    "391 \"Thoubal\"\n",
    "392 \"Ukhrul\"\n",
    "393 \"East Garo Hills\"\n",
    "394 \"East Jaintia Hills\"\n",
    "395 \"East Khasi Hills\"\n",
    "396 \"North Garo Hills\"\n",
    "397 \"Ri Bhoi\"\n",
    "398 \"South Garo Hills\"\n",
    "399 \"South West Garo Hills\"\n",
    "400 \"South West Khasi Hills\"\n",
    "401 \"West Garo Hills\"\n",
    "402 \"West Jaintia Hills\"\n",
    "403 \"West Khasi Hills\"\n",
    "404 \"Aizawl\"\n",
    "405 \"Champhai\"\n",
    "406 \"Kolasib\"\n",
    "407 \"Lawngtlai\"\n",
    "408 \"Lunglei\"\n",
    "409 \"Mamit\"\n",
    "410 \"Saiha\"\n",
    "411 \"Serchhip\"\n",
    "412 \"Dimapur\"\n",
    "413 \"Kiphire\"\n",
    "414 \"Kohima\"\n",
    "415 \"Longleng\"\n",
    "416 \"Mokokchung\"\n",
    "417 \"Mon\"\n",
    "418 \"Peren\"\n",
    "419 \"Phek\"\n",
    "420 \"Tuensang\"\n",
    "421 \"Wokha\"\n",
    "422 \"Zunheboto\"\n",
    "423 \"Angul\"\n",
    "424 \"Balangir\"\n",
    "425 \"Balasore\"\n",
    "426 \"Bargarh (Baragarh)\"\n",
    "427 \"Bhadrak\"\n",
    "428 \"Boudh (Bauda)\"\n",
    "429 \"Cuttack\"\n",
    "430 \"Debagarh (Deogarh)\"\n",
    "431 \"Dhenkanal\"\n",
    "432 \"Gajapati\"\n",
    "433 \"Ganjam\"\n",
    "434 \"Jagatsinghpur\"\n",
    "435 \"Jajpur\"\n",
    "436 \"Jharsuguda\"\n",
    "437 \"Kalahandi\"\n",
    "438 \"Kandhamal\"\n",
    "439 \"Kendrapara\"\n",
    "440 \"Kendujhar (Keonjhar)\"\n",
    "441 \"Khordha\"\n",
    "442 \"Koraput\"\n",
    "443 \"Malkangiri\"\n",
    "444 \"Mayurbhanj\"\n",
    "445 \"Nabarangpur\"\n",
    "446 \"Nayagarh\"\n",
    "447 \"Nuapada\"\n",
    "448 \"Puri\"\n",
    "449 \"Rayagada\"\n",
    "450 \"Sambalpur\"\n",
    "451 \"Subarnapur (Sonepur)\"\n",
    "452 \"Sundargarh\"\n",
    "453 \"Karaikal\"\n",
    "454 \"Mahe\"\n",
    "455 \"Pondicherry\"\n",
    "456 \"Yanam\"\n",
    "457 \"Amritsar\"\n",
    "458 \"Barnala\"\n",
    "459 \"Bathinda\"\n",
    "460 \"Faridkot\"\n",
    "461 \"Fatehgarh Sahib\"\n",
    "462 \"Fazilka\"\n",
    "463 \"Firozpur\"\n",
    "464 \"Gurdaspur\"\n",
    "465 \"Hoshiarpur\"\n",
    "466 \"Jalandhar\"\n",
    "467 \"Kapurthala\"\n",
    "468 \"Ludhiana\"\n",
    "469 \"Mansa\"\n",
    "470 \"Moga\"\n",
    "471 \"Pathankot\"\n",
    "472 \"Patiala\"\n",
    "473 \"Rupnagar\"\n",
    "474 \"Sahibzada Ajit Singh Nagar(Mohali)\"\n",
    "475 \"Sangrur\"\n",
    "476 \"Shahid Bhagat Singh Nagar\"\n",
    "477 \"Sri Muktsar Sahib\"\n",
    "478 \"Tarn Taran\"\n",
    "479 \"Ajmer\"\n",
    "480 \"Alwar\"\n",
    "481 \"Banswara\"\n",
    "482 \"Baran\"\n",
    "483 \"Barmer\"\n",
    "484 \"Bharatpur\"\n",
    "485 \"Bhilwara\"\n",
    "486 \"Bikaner\"\n",
    "487 \"Bundi\"\n",
    "488 \"Chittorgarh\"\n",
    "489 \"Churu\"\n",
    "490 \"Dausa\"\n",
    "491 \"Dholpur\"\n",
    "492 \"Dungapur\"\n",
    "493 \"Ganganagar\"\n",
    "494 \"Hanumangarh\"\n",
    "495 \"Jaipur\"\n",
    "496 \"Jaisalmer\"\n",
    "497 \"Jalore\"\n",
    "498 \"Jhalawar\"\n",
    "499 \"Jhunjhunu\"\n",
    "500 \"Jodhpur\"\n",
    "501 \"Karauli\"\n",
    "502 \"Kota\"\n",
    "503 \"Nagaur\"\n",
    "504 \"Pali\"\n",
    "505 \"Pratapgarh\"\n",
    "506 \"Rajsamand\"\n",
    "507 \"Sawai Madhopur\"\n",
    "508 \"Sikar\"\n",
    "509 \"Sirohi\"\n",
    "510 \"Tonk\"\n",
    "511 \"Udaipur\"\n",
    "512 \"East Sikkim\"\n",
    "513 \"North Sikkim\"\n",
    "514 \"South Sikkim\"\n",
    "515 \"West Sikkim\"\n",
    "516 \"Ariyalur\"\n",
    "517 \"Chennai\"\n",
    "518 \"Coimbatore\"\n",
    "519 \"Cuddalore\"\n",
    "520 \"Dharmapuri\"\n",
    "521 \"Dindigul\"\n",
    "522 \"Erode\"\n",
    "523 \"Kanchipuram\"\n",
    "524 \"Kanyakumari\"\n",
    "525 \"Karur\"\n",
    "526 \"Krishnagiri\"\n",
    "527 \"Madurai\"\n",
    "528 \"Nagapattinam\"\n",
    "529 \"Namakkal\"\n",
    "530 \"Nilgiris\"\n",
    "531 \"Perambalur\"\n",
    "532 \"Pudukkottai\"\n",
    "533 \"Ramanathapuram\"\n",
    "534 \"Salem\"\n",
    "535 \"Sivaganga\"\n",
    "536 \"Thanjavur\"\n",
    "537 \"Theni\"\n",
    "538 \"Thoothukudi\"\n",
    "539 \"Tiruchirappalli\"\n",
    "540 \"Tirunelveli\"\n",
    "541 \"Tirupur\"\n",
    "542 \"Tiruvallur\"\n",
    "543 \"Tiruvannamalai\"\n",
    "544 \"Tiruvarur\"\n",
    "545 \"Vellore\"\n",
    "546 \"Viluppuram\"\n",
    "547 \"Virudhunagar\"\n",
    "548 \"Adilabad\"\n",
    "549 \"Hyderabad\"\n",
    "550 \"Karimnagar\"\n",
    "551 \"Khammam\"\n",
    "552 \"Mahbubnagar\"\n",
    "553 \"Medak\"\n",
    "554 \"Nalgonda\"\n",
    "555 \"Nizamabad\"\n",
    "556 \"Ranga Reddy\"\n",
    "557 \"Warangal\"\n",
    "558 \"Dhalai\"\n",
    "559 \"Gomati\"\n",
    "560 \"Khowai\"\n",
    "561 \"North Tripura\"\n",
    "562 \"Sepahijala\"\n",
    "563 \"South Tripura\"\n",
    "564 \"Unokoti\"\n",
    "565 \"West Tripura\"\n",
    "566 \"Agra\"\n",
    "567 \"Aligarh\"\n",
    "568 \"Allahabad\"\n",
    "569 \"Ambedkar Nagar\"\n",
    "570 \"Amethi (Chhatrapati Shahuji Maharaj Nagar)\"\n",
    "571 \"Amroha (Jyotiba Phule Nagar)\"\n",
    "572 \"Auraiya\"\n",
    "573 \"Azamgarh\"\n",
    "574 \"Bagpat\"\n",
    "575 \"Bahraich\"\n",
    "576 \"Ballia\"\n",
    "577 \"Balrampur\"\n",
    "578 \"Banda\"\n",
    "579 \"Barabanki\"\n",
    "580 \"Bareilly\"\n",
    "581 \"Basti\"\n",
    "582 \"Bijnor\"\n",
    "583 \"Budaun\"\n",
    "584 \"Bulandshahr\"\n",
    "585 \"Chandauli\"\n",
    "586 \"Chitrakoot\"\n",
    "587 \"Deoria\"\n",
    "588 \"Etah\"\n",
    "589 \"Etawah\"\n",
    "590 \"Faizabad\"\n",
    "591 \"Farrukhabad\"\n",
    "592 \"Fatehpur\"\n",
    "593 \"Firozabad\"\n",
    "594 \"Gautam Buddh Nagar\"\n",
    "595 \"Ghaziabad\"\n",
    "596 \"Ghazipur\"\n",
    "597 \"Gonda\"\n",
    "598 \"Gorakhpur\"\n",
    "599 \"Hamirpur\"\n",
    "600 \"Hapur (Panchsheel Nagar)\"\n",
    "601 \"Hathras (Mahamaya Nagar)\"\n",
    "602 \"Jalaun\"\n",
    "603 \"Jaunpur district\"\n",
    "604 \"Jhansi\"\n",
    "605 \"Kannauj\"\n",
    "606 \"Kanpur Dehat (Ramabai Nagar)\"\n",
    "607 \"Kanpur Nagar\"\n",
    "608 \"Kasganj (Kanshi Ram Nagar)\"\n",
    "609 \"Kaushambi\"\n",
    "610 \"Kushinagar\"\n",
    "611 \"Lakhimpur Kheri\"\n",
    "612 \"Lalitpur\"\n",
    "613 \"Lucknow\"\n",
    "614 \"Maharajganj\"\n",
    "615 \"Mahoba\"\n",
    "616 \"Mainpuri\"\n",
    "617 \"Mathura\"\n",
    "618 \"Mau\"\n",
    "619 \"Meerut\"\n",
    "620 \"Mirzapur\"\n",
    "621 \"Moradabad\"\n",
    "622 \"Muzaffarnagar\"\n",
    "623 \"Pilibhit\"\n",
    "624 \"Pratapgarh\"\n",
    "625 \"Raebareli\"\n",
    "626 \"Rampur\"\n",
    "627 \"Saharanpur\"\n",
    "628 \"Sambhal (Bheem Nagar)\"\n",
    "629 \"Sant Kabir Nagar\"\n",
    "630 \"Shahjahanpur\"\n",
    "631 \"Shamli\"\n",
    "632 \"Shravasti\"\n",
    "633 \"Siddharthnagar\"\n",
    "634 \"Sitapur\"\n",
    "635 \"Sonbhadra\"\n",
    "636 \"Sultanpur\"\n",
    "637 \"Unnao\"\n",
    "638 \"Varanasi\"\n",
    "639 \"Almora\"\n",
    "640 \"Bageshwar\"\n",
    "641 \"Chamoli\"\n",
    "642 \"Champawat\"\n",
    "643 \"Dehradun\"\n",
    "644 \"Haridwar\"\n",
    "645 \"Nainital\"\n",
    "646 \"Pauri Garhwal\"\n",
    "647 \"Pithoragarh\"\n",
    "648 \"Rudraprayag\"\n",
    "649 \"Tehri Garhwal\"\n",
    "650 \"Udham Singh Nagar\"\n",
    "651 \"Uttarkashi\"\n",
    "652 \"Alipurduar\"\n",
    "653 \"Bankura\"\n",
    "654 \"Purba Bardhaman\"\n",
    "655 \"Birbhum\"\n",
    "656 \"Cooch Behar\"\n",
    "657 \"Dakshin Dinajpur\"\n",
    "658 \"Darjeeling\"\n",
    "659 \"Hooghly\"\n",
    "660 \"Howrah\"\n",
    "661 \"Jalpaiguri\"\n",
    "662 \"Kolkata\"\n",
    "663 \"Maldah\"\n",
    "664 \"Murshidabad\"\n",
    "665 \"Nadia\"\n",
    "666 \"North 24 Parganas\"\n",
    "667 \"Paschim Medinipur\"\n",
    "668 \"Purba Medinipur\"\n",
    "669 \"Purulia\"\n",
    "670 \"South 24 Parganas\"\n",
    "671 \"Uttar Dinajpur\"\n",
    "672 \"Kalimpong\"\n",
    "673 \"Jhargram\"\n",
    "674 \"Paschim Bardhaman\"\n",
    "675 \"Chhapra\"\n",
    "676 \"Hardoi\"\n",
    "677 \"Charkhi Dadri\"\n",
    "678 \"Hojai\"\n",
    "'''\n",
    "\n",
    "districts = {}\n",
    "\n",
    "for line in text6.strip().split(\"\\n\"):\n",
    "    parts = line.split(\" \", 1) \n",
    "    key = int(parts[0])\n",
    "    value = parts[1].strip('\"')\n",
    "    districts[key] = value\n",
    "\n",
    "df['District'] = df['District'].replace(districts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "792eccf7-2884-4e48-8cf6-93ca9d8bb7b7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 5185 entries, 0 to 5184\n",
      "Data columns (total 34 columns):\n",
      " #   Column                                         Non-Null Count  Dtype         \n",
      "---  ------                                         --------------  -----         \n",
      " 0   AMR ID                                         5185 non-null   int32         \n",
      " 1   Suscepbility ID                                5185 non-null   int32         \n",
      " 2   Sample ID                                      5185 non-null   int32         \n",
      " 3   Hospital Patient Relation ID                   5185 non-null   int32         \n",
      " 4   Patient ID                                     5185 non-null   int32         \n",
      " 5   Antibiotic                                     4817 non-null   object        \n",
      " 6   Panel ID                                       5185 non-null   object        \n",
      " 7   special test type                              368 non-null    object        \n",
      " 8   Special Test                                   368 non-null    object        \n",
      " 9   Organism                                       5185 non-null   object        \n",
      " 10  Identification Method                          5169 non-null   object        \n",
      " 11  Susceptibility Test Type                       4817 non-null   object        \n",
      " 12  Value                                          5185 non-null   object        \n",
      " 13  Remarks                                        4805 non-null   object        \n",
      " 14  Gender                                         5185 non-null   object        \n",
      " 15  Age                                            5185 non-null   int8          \n",
      " 16  District                                       5139 non-null   object        \n",
      " 17  State                                          5139 non-null   float64       \n",
      " 18  Country                                        5185 non-null   int8          \n",
      " 19  location type                                  3690 non-null   float64       \n",
      " 20  Hospital ID                                    5185 non-null   int8          \n",
      " 21  Date of Admission                              5185 non-null   datetime64[ns]\n",
      " 22  Hospital department                            5185 non-null   int16         \n",
      " 23  Hospital Department type                       5185 non-null   int8          \n",
      " 24  Infection Type                                 5185 non-null   int8          \n",
      " 25  Clinical Diagnosis                             5185 non-null   object        \n",
      " 26  Comorbidity                                    5185 non-null   object        \n",
      " 27  Whether device used?                           5185 non-null   int8          \n",
      " 28  Patient with Devices                           5185 non-null   object        \n",
      " 29  Prior Admission in Last 3 months               240 non-null    float64       \n",
      " 30  Date of sample Collection                      5185 non-null   datetime64[ns]\n",
      " 31  Sample Type                                    5185 non-null   int8          \n",
      " 32  antibiotic prescription in last 1 months?      5185 non-null   int8          \n",
      " 33  Antibiotics/Antifungals taken in last 1 month  5185 non-null   object        \n",
      "dtypes: datetime64[ns](2), float64(3), int16(1), int32(5), int8(8), object(15)\n",
      "memory usage: 962.2+ KB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "d7ae855e-9878-4efc-8e18-1f2b0da448a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "text7 = '''1 \"Andaman and Nicobar\"\n",
    "2 \"Andhra Pradesh\"\n",
    "3 \"Arunachal Pradesh\"\n",
    "4 \"Assam\"\n",
    "5 \"Bihar\"\n",
    "6 \"Chandigarh\"\n",
    "7 \"Chhattisgarh\"\n",
    "8 \"Dadra and Nagar Haveli\"\n",
    "9 \"Daman and Diu\"\n",
    "10 \"Delhi\"\n",
    "11 \"Goa\"\n",
    "12 \"Gujarat\"\n",
    "13 \"Haryana\"\n",
    "14 \"Himachal Pradesh\"\n",
    "15 \"Jammu and Kashmir\"\n",
    "16 \"Jharkhand\"\n",
    "17 \"Karnataka\"\n",
    "18 \"Kerala\"\n",
    "19 \"Ladakh\"\n",
    "20 \"Lakshadweep (LD)\"\n",
    "21 \"Madhya Pradesh\"\n",
    "22 \"Maharashtra\"\n",
    "23 \"Manipur\"\n",
    "24 \"Meghalaya\"\n",
    "25 \"Mizoram\"\n",
    "26 \"Nagaland\"\n",
    "27 \"Odisha\"\n",
    "28 \"Puducherry\"\n",
    "29 \"Punjab\"\n",
    "30 \"Rajasthan\"\n",
    "31 \"Sikkim\"\n",
    "32 \"Tamil Nadu\"\n",
    "33 \"Telangana\"\n",
    "34 \"Tripura\"\n",
    "35 \"Uttar Pradesh\"\n",
    "36 \"Uttarakhand\"\n",
    "37 \"West Bengal\"\n",
    "'''\n",
    "\n",
    "states = {}\n",
    "\n",
    "for line in text7.strip().split(\"\\n\"):\n",
    "    parts = line.split(\" \", 1) \n",
    "    key = int(parts[0])\n",
    "    value = parts[1].strip('\"')\n",
    "    states[key] = value\n",
    "\n",
    "df['State'] = df['State'].replace(states)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "c771c25b-3041-40bc-981d-85269107a964",
   "metadata": {},
   "outputs": [],
   "source": [
    "def text_to_dict_and_replace(df, column_name, text):\n",
    "    \"\"\"\n",
    "    Converts a multi-line string of numeric keys and string values into a dictionary\n",
    "    and replaces numeric values in a specified DataFrame column.\n",
    "\n",
    "    Args:\n",
    "        df (pd.DataFrame): The DataFrame where replacements will be applied.\n",
    "        column_name (str): The column in which numeric values should be replaced.\n",
    "        text (str): Multiline string containing numeric keys and corresponding string values.\n",
    "\n",
    "    Returns:\n",
    "        pd.DataFrame: DataFrame with replaced values.\n",
    "        dict: The mapping dictionary.\n",
    "    \"\"\"\n",
    "    mapping_dict = {}\n",
    "\n",
    "    # Convert text to dictionary\n",
    "    for line in text.strip().split(\"\\n\"):\n",
    "        parts = line.split(\" \", 1)  # Split on the first space\n",
    "        if len(parts) == 2:\n",
    "            key = int(parts[0])  # Convert first part to integer\n",
    "            value = parts[1].strip('\"')  # Remove surrounding quotes\n",
    "            mapping_dict[key] = value  # Store in dictionary\n",
    "\n",
    "    # Replace numeric values in the DataFrame\n",
    "    df[column_name] = df[column_name].replace(mapping_dict)\n",
    "\n",
    "    #return df, mapping_dict  # Return modified DataFrame and dictionary\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "e65a6a20-8de5-4589-b487-897c9f573065",
   "metadata": {},
   "outputs": [],
   "source": [
    "text8 = '''1 \"India\"\n",
    "2 \"Outside India\"\n",
    "'''\n",
    "text_to_dict_and_replace(df, 'Country', text8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "82615bcf-2690-4eb0-9340-2e5597d445c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "text9 = '''1 \"Rural\"\n",
    "2 \"Urban\"\n",
    "'''\n",
    "text_to_dict_and_replace(df, 'location type', text9)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "bd563d11-ea6d-457e-894b-1872a9a5cc20",
   "metadata": {},
   "outputs": [],
   "source": [
    "text10 = '''1 \"Cardiology\"\n",
    "2 \"Dermatology\"\n",
    "3 \"Medicine\"\n",
    "4 \"Gastroenterology\"\n",
    "5 \"Hematology\"\n",
    "6 \"Hepatology\"\n",
    "7 \"Nephrology\"\n",
    "8 \"Neurology\"\n",
    "10 \"Pulmonary medicine\"\n",
    "11 \"Psychiatry\"\n",
    "12 \"Oral health (Dentistry)\"\n",
    "13 \"Urology\"\n",
    "28 \"General surgery\"\n",
    "35 \"Neurosurgery\"\n",
    "38 \"Truma unit\"\n",
    "39 \"Otolaryngology (ENT)\"\n",
    "41 \"Plastic surgery\"\n",
    "42 \"Renal transplant\"\n",
    "44 \"Special clinic\"\n",
    "45 \"Anatomy\"\n",
    "46 \"Biochemistry\"\n",
    "47 \"College of Nursing\"\n",
    "48 \"Pathology\"\n",
    "49 \"Pharmacology\"\n",
    "50 \"Forensic Medicine and Toxicology\"\n",
    "51 \"Medical Education\"\n",
    "52 \"Medical Biometrics and Informatics (Biostatistics)\"\n",
    "53 \"Physiology\"\n",
    "54 \"Microbiology\"\n",
    "55 \"Preventive and Social Medicine\"\n",
    "57 \"Obstetrics and Gynaecology\"\n",
    "58 \"Ophthalmology\"\n",
    "59 \"Orthopaedic Surgery\"\n",
    "60 \"Pediatric Surgery\"\n",
    "62 \"Surgical Gastroenterology\"\n",
    "63 \"Surgical Oncology\"\n",
    "64 \"Surgery\"\n",
    "67 \"Cardio thoracic and Vascular Surgery\"\n",
    "68 \"Anaesthesiology & Critical Care\"\n",
    "71 \"Dermatology and STD\"\n",
    "72 \"Endocrinology\"\n",
    "73 \"Medical Gastroenterology\"\n",
    "74 \"Medical Oncology\"\n",
    "77 \"Nuclear Medicine\"\n",
    "79 \"Pediatric\"\n",
    "82 \"Radiotherapy\"\n",
    "83 \"Radio diagnosis\"\n",
    "84 \"Transfusion Medicine\"\n",
    "85 \"Physical Medicine and Rehabilitation (PMR)\"\n",
    "86 \"Clinical Immunology\"\n",
    "87 \"Clinical Pharmacology\"\n",
    "88 \"Emergency Medicine and Trauma\"\n",
    "89 \"Neonatology\"\n",
    "93 \"G.I. Surgery\"\n",
    "94 \"Geriatric medicine\"\n",
    "95 \"IRCH (Malignancy)\"\n",
    "96 \"Advance Trauma Centre\"\n",
    "97 \"Anaesthesia and Intensive Care\"\n",
    "98 \"Bone Marrow Transplant\"\n",
    "99 \"Burn Unit\"\n",
    "101 \"Communicable Disease Ward\"\n",
    "102 \"Drug De-Addiction and Treatment\"\n",
    "103 \"Emergency Medicine\"\n",
    "104 \"Emergency Surgery\"\n",
    "107 \"High Dependency Unit (HDU)\"\n",
    "108 \"Internal Medicine\"\n",
    "114 \"Orthopedics\"\n",
    "116 \"Pediatric Medicine\"\n",
    "117 \"Pediatric Orthopedics\"\n",
    "120 \"Pulmonology\"\n",
    "123 \"Others\"\n",
    "124 \"Accident and Emergency\"\n",
    "125 \"Anaesthesia\"\n",
    "127 \"Cardio thoracic surgery\"\n",
    "128 \"Child health\"\n",
    "129 \"Community health and Development\"\n",
    "130 \"Dental and Oral surgery\"\n",
    "134 \"Family medicine\"\n",
    "136 \"Geriatrics\"\n",
    "137 \"Gynaecologic oncology\"\n",
    "139 \"Hand and Leprosy reconstructive surgery\"\n",
    "140 \"Low Cost Effective Care Unit (LCECU)\"\n",
    "145 \"Neurosciences\"\n",
    "152 \"Reproductive medicine\"\n",
    "153 \"Respiratory medicine\"\n",
    "154 \"Rheumatology\"\n",
    "155 \"Rural Unit for Health and Social Affairs (RUHSA)\"\n",
    "156 \"Staff student health service\"\n",
    "157 \"Breast Oncology\"\n",
    "158 \"Urological Oncology\"\n",
    "159 \"Head and Neck Surgical Oncology\"\n",
    "160 \"Gastro-intestinal and hepatobiliary surgical oncology\"\n",
    "161 \"Pediatric Oncology\"\n",
    "162 \"Clinical Nutrition\"\n",
    "'''\n",
    "text_to_dict_and_replace(df, 'Hospital department', text10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "974f7dcd-61a5-474a-88f0-28ec898dcbb5",
   "metadata": {},
   "outputs": [],
   "source": [
    "text11 = '''1 \"ICU\"\n",
    "2 \"OPD\"\n",
    "3 \"Ward\"\n",
    "'''\n",
    "text_to_dict_and_replace(df, 'Hospital Department type', text11)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "3dbc4dd5-dc2a-41b1-bb9d-4889860a553e",
   "metadata": {},
   "outputs": [],
   "source": [
    "text12 = '''1 \"Community Acquired Infection\"\n",
    "2 \"Health Care Associated\" Infection\n",
    "3 \"Not Known\"\n",
    "'''\n",
    "text_to_dict_and_replace(df, 'Infection Type', text12)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "d1925bdd-67b0-4301-a169-df61193f62aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "text13 = '''0 \"no\"\n",
    "1 \"yes\"\n",
    "'''\n",
    "text_to_dict_and_replace(df, 'Whether device used?', text13)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "a12168f9-5451-4bdf-be19-c20f6d7c1833",
   "metadata": {},
   "outputs": [],
   "source": [
    "text14 = '''1 \"ICU\"\n",
    "2 \"Ward\"\n",
    "'''\n",
    "text_to_dict_and_replace(df, 'Prior Admission in Last 3 months', text14)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "55565995-b040-465f-a5eb-d280d698e081",
   "metadata": {},
   "outputs": [],
   "source": [
    "text15 = '''1 \"BAL\"\n",
    "3 \"Sputum\"\n",
    "4 \"Deep Biopsy\"\n",
    "5 \"Blood-central catheter\"\n",
    "6 \"Blood-peripheral\"\n",
    "7 \"CSF\"\n",
    "8 \"Fluid from sterile site\"\n",
    "9 \"Genital tract\"\n",
    "10 \"Ocular-superficial\"\n",
    "11 \"Ocular-deep\"\n",
    "12 \"Pus/exudate\"\n",
    "13 \"Faeces\"\n",
    "14 \"Throat swab\"\n",
    "15 \"Urine\"\n",
    "16 \"Wound swab\"\n",
    "17 \"Abdominal fluid\"\n",
    "18 \"Abscess aspirate\"\n",
    "20 \"Bile\"\n",
    "21 \"Brain abscess\"\n",
    "22 \"Lung aspirate\"\n",
    "23 \"Endotracheal aspirate (ETA)\"\n",
    "24 \"Ethmoidal tissue\"\n",
    "25 \"FNAC\"\n",
    "27 \"Gastric fluid\"\n",
    "28 \"ICTD fluid\"\n",
    "29 \"Intestinal fluid\"\n",
    "30 \"Intra operative bile\"\n",
    "31 \"Intra operative fluid\"\n",
    "32 \"Intracostal tube fluid\"\n",
    "34 \"Joint Fluid\"\n",
    "35 \"Iobectomy tissue\"\n",
    "36 \"Maxillary alveolar tissue\"\n",
    "38 \"Maxillary sinus\"\n",
    "44 \"Nasal scrapping\"\n",
    "45 \"Nasal tissue\"\n",
    "46 \"Necrosum\"\n",
    "47 \"Needle aspirate\"\n",
    "48 \"Oral cavity crust\"\n",
    "49 \"Palatal mucosa\"\n",
    "50 \"Palatal scrapping\"\n",
    "52 \"Pancreatic drain fluid\"\n",
    "53 \"Pericardial fluid\"\n",
    "54 \"Peripheral catheter-blood\"\n",
    "55 \"Peritoneal fluid\"\n",
    "56 \"Pigtail fluid\"\n",
    "57 \"Pleural fluid\"\n",
    "58 \"PTBD fluid\"\n",
    "60 \"Sinus mass\"\n",
    "62 \"Skin biopsy\"\n",
    "63 \"Deep Tissue\"\n",
    "64 \"Vitreous fluid\"\n",
    "65 \"Cervical Swab\"\n",
    "66 \"Drain Fluid\"\n",
    "67 \"Placental Membrane\"\n",
    "68 \"Semen\"\n",
    "69 \"Ear Swab\"\n",
    "70 \"Bone\"\n",
    "71 \"Nail\"\n",
    "72 \"Hair\"\n",
    "73 \"Instrument\"\n",
    "74 \"Vaginal swab\"\n",
    "75 \"Others\"\n",
    "76 \"Bone Marrow\"\n",
    "77 \"Vascular Catheter tip\"\n",
    "78 \"Endometrium\"\n",
    "79 \"Lymph node\"\n",
    "80 \"Skin and Soft Tissue\"\n",
    "81 \"Superficial Biopsy\"\n",
    "82 \"Superficial Tissue\"\n",
    "83 \"Pus aspirate\"\n",
    "84 \"Abdominal Tissue\"\n",
    "85 \"Abdominal Pus\"\n",
    "'''\n",
    "text_to_dict_and_replace(df, 'Sample Type', text15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "9f10beb9-5229-4145-b032-b32f42567e88",
   "metadata": {},
   "outputs": [],
   "source": [
    "text16 = '''0 \"no\"\n",
    "1 \"yes\"\n",
    "'''\n",
    "text_to_dict_and_replace(df, 'antibiotic prescription in last 1 months?', text16)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "1f444796-6a21-4877-aa4d-21253a8fad00",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('ICMR Antimicrobial Resistance.csv', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dee2c827-b97f-41b3-ac1a-f10aadf26219",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e56e9af-16e1-4aba-a72a-0227d2a2faaa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bce786dd-57a3-42cd-b7b2-fe5c72f86ab6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c5970c1b-c891-405d-8970-7fe6c8da1a72",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e1fb850b-7f9b-4de2-a9d2-cf6531b2c174",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
